packages/oid4vci-demo-frontend/src/components/Form/index.tsx

```diff
@@ -5,15 +5,16 @@ import {DataFormElement, DataFormRow} from '../../ecosystem/ecosystem-config'
 import {generateRandomIBAN} from '../../utils/iban'
 import {transformFormConfigToEmptyObject} from '../../utils/ObjectUtils'
 import InputField from '../InputField'
-import {FormData, FormFieldValue} from '../../types'
+import {FormOutputData, FormFieldValue, ImmutableRecord} from '../../types'
 import style from './index.module.css'
 
 type Props = {
-    form: DataFormRow[]
-    onChange?: (formData: FormData) => Promise<void>
+    formConfig: DataFormRow[]
+    formInitData?: ImmutableRecord
```

Comment:
Maybe this should not be a property, but it's what's easy to access in a tsx.
To me it makes more sense to have three separate elements, the config is the config from the json which should be immutable. Then the initial data / the current record to pre-load the form with. And finally the form output.
For Bram it makes more sense to merge the config with the the initial data / the current record into one prop.
Guess we need a referee :)

Reply:
@nklomp ^^^ what is your opinion?

Reply:
Also it's maybe better to have both read-only and editable properties. read-only is always read-only, editable only read-only when the field has initial data

Reply:
Separate for sure, but at the same time, this is not a product, although we obviously will use parts as inspiration in VDX

Reply:
i'm fine with this as indeed it's not a product, but my argument against this would be that with this change you can have a missmatch between the config and the initial data. The way i build the form is to be as dumb as possible and only have it use 1 prop to configure the form. Yes you need to maintain a record somewhere obv for the current state but this is internally and the onChange should be used to expose it to the outside.